GitCommands/Git/RevisionDiffProvider.cs

```diff
@@ -13,6 +13,18 @@ public interface IRevisionDiffProvider
         /// <param name="to">The second "current" revision</param>
         /// <returns></returns>
         string Get(string from, string to);
+
+        /// <summary>
+        /// options to git-diff from GE arguments, including artificial commits
+        /// This is an instance class to not have static dependencies in GitModule
+        /// </summary>
+        /// <param name="from">The first revision</param>
+        /// <param name="to">The second "current" revision</param>
+        /// <param name="fileName">The file to compare</param>
+        /// <param name="oldFileName">The old name of the file</param>
+        /// <param name="isTracked">The file is tracked</param>
+        /// <returns></returns>
+        string Get(string from, string to, string fileName, string oldFileName, bool isTracked);
```

Comment:
The order of parameters looks a little odd to me - `from, to` = past->future, `filename, oldFileName` = current->past
Perhaps we can rearrange the parameters to `from, oldFileName, to, currentFileName, isTracked`

Perhaps even rename `to` and `from` to something a bit more meaningful - like `toHash` or `toSha1`. 
If the type was `GitRevisions` it could be relatively obvious what `to` and `from` were, however when it is a string it is not so obvious....

Reply:
I changed the names in the commit but believe the original names are better: the arguments corresponds to git diff syntax.

Reply:
the man page mentions A-commit and B-commit as well as first and second commit to view the changes from one commit to another. 
GE uses revision rather than commit, so that should be replaced. A and B is used in the UI.

The most correct would probably be A-revision, with naming rules would be aRevision, which I read as "a revision".

What about revA and revB?

Reply:
Regarding GitRevisions vs string: I tried to use that (RevisionDiffProvider.Get() was part of GitRevision originally). In many situations it would have been possible, but in a little too many a separate GitRevision would have had to be created, which I believe was too complex. (The artificial diff PRs was quite big anyway.)

Reply:
> The order is also similar to the parameters to git-diff.

That's cool with me then

> Regarding GitRevisions vs string

My comment was about readability. Whether the current API is as intuitive as it can be, or whether we can make it better.